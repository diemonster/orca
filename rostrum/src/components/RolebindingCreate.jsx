import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';

import {
  rolebindingCreate,
  rolebindingCreateChangeInput,
} from '../actions/rolebindingActions';


export class RolebindingCreate extends React.Component {
  constructor(props) {
    super(props);

    this.handleChange = this.handleChange.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
  }

  handleChange(event) {
    event.preventDefault();
    const { dispatchRolebindingCreateChangeInput } = this.props;
    const inputType = event.target.name;
    const inputValue = event.target.value;
    dispatchRolebindingCreateChangeInput(inputType, inputValue);
  }

  handleSubmit(event) {
    event.preventDefault();
    const {
      dispatchRolebindingCreate,
      rolebindingCreateRoleInput,
      rolebindingCreateSubjectInput,
      selectedNamespace,
    } = this.props;

    dispatchRolebindingCreate(
      selectedNamespace,
      rolebindingCreateRoleInput,
      rolebindingCreateSubjectInput,
    );
  }

  render() {
    const {
      rolebindingCreateSubjectInput,
      rolebindingCreateRoleInput,
      roleOptions,
    } = this.props;

    const rolebindingOptions = roleOptions.map(option => (
      <option key={option} value={option}>{option}</option>
    ));

    return (
      <div className="create-rolebinding">
        <h2>Create New Rolebinding</h2>
        <form className="rolebinding-input-form" id="rolebinding-create-input-form" onSubmit={this.handleSubmit}>
          <input
            type="text"
            name="subject"
            value={rolebindingCreateSubjectInput}
            onChange={this.handleChange}
            placeholder="user"
            id="rolebinding-create-subject-input"
          />
          <select
            name="role"
            required
            value={rolebindingCreateRoleInput}
            onChange={this.handleChange}
          >
            {rolebindingOptions}
          </select>
          <button
            className="button"
            type="submit"
            value="submit"
          >
            Create
          </button>
        </form>
      </div>
    );
  }
}

RolebindingCreate.propTypes = {
  dispatchRolebindingCreate: PropTypes.func.isRequired,
  dispatchRolebindingCreateChangeInput: PropTypes.func.isRequired,
  rolebindingCreateRoleInput: PropTypes.string.isRequired,
  rolebindingCreateSubjectInput: PropTypes.string.isRequired,
  roleOptions: PropTypes.arrayOf(PropTypes.string).isRequired,
  selectedNamespace: PropTypes.string.isRequired,
};

const mapStateToProps = state => ({
  rolebindingCreateRoleInput: state.rolebinding.roleInput,
  rolebindingCreateSubjectInput: state.rolebinding.subjectInput,
  roleOptions: state.rolebinding.roleOptions,
  selectedNamespace: state.namespace.selectedNamespace,
});

const mapDispatchToProps = dispatch => ({
  dispatchRolebindingCreate: (namespace, role, subject) => {
    dispatch(rolebindingCreate(namespace, role, subject));
  },
  dispatchRolebindingCreateChangeInput: (inputType, inputValue) => {
    dispatch(rolebindingCreateChangeInput(inputType, inputValue));
  },
});

export default connect(mapStateToProps, mapDispatchToProps)(RolebindingCreate);
