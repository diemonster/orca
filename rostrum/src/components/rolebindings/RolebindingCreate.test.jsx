import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import { RolebindingCreate } from './RolebindingCreate';

Enzyme.configure({ adapter: new Adapter() });

describe('components', () => {
  describe('RolebindingCreate', () => {
    it('should render itself and its subcomponents', () => {
      const props = {
        dispatchRolebindingCreate: jest.fn(),
        dispatchRolebindingCreateChangeInput: jest.fn(),
        rolebindingCreateRoleInput: '',
        rolebindingCreateSubjectInput: '',
        roleOptions: [],
        selectedNamespace: '',
      };

      const rolebindingCreate = shallow(<RolebindingCreate {...props} />);
      expect(rolebindingCreate).toMatchSnapshot();
    });
  });
});
