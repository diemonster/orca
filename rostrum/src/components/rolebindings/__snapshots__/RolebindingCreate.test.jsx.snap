// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components RolebindingCreate should render itself and its subcomponents 1`] = `
<div
  className="create-rolebinding"
>
  <h2>
    Create New Rolebinding
  </h2>
  <form
    className="rolebinding-input-form"
    id="rolebinding-create-input-form"
    onSubmit={[Function]}
  >
    <input
      id="rolebinding-create-subject-input"
      name="subject"
      onChange={[Function]}
      placeholder="user"
      type="text"
      value=""
    />
    <select
      name="role"
      onChange={[Function]}
      required={true}
      value=""
    >
      <option />
    </select>
    <button
      className="button"
      type="submit"
      value="submit"
    >
      Create
    </button>
  </form>
</div>
`;
